packages/react-components/react-select/src/components/Select/useSelectStyles.ts

```diff
@@ -108,25 +110,59 @@ const useSelectStyles = makeStyles({
       ...shorthands.borderColor('GrayText'),
     },
   },
+  /* The right padding is calculated  as the sum of the following:
+   * 1. Field padding-right
+   * 2. Icon width
+   * 3. Content-icon spacing
+   * 4. Content inner padding
+   */
   small: {
     height: fieldHeights.small,
-    ...shorthands.padding('0', tokens.spacingHorizontalSNudge),
+    paddingBottom: 0,
+    paddingLeft: `calc(${tokens.spacingHorizontalSNudge} + ${tokens.spacingHorizontalXXS})`,
+    paddingRight: `calc(${tokens.spacingHorizontalSNudge}
```

Comment:
this feels like a LOT of calculations to get padding. is there some other way to create this space? is our layout approach not sufficient for this? Do we need to switch to grid or something?

Reply:
it's b/c it's a `<select>` element, so the caret is absolutely positioned over the actual `<select>`, and can't be a child of it. I'm not sure if there's really a better way to do it :/